fastlane/lib/fastlane/actions/zip.rb

```diff
@@ -0,0 +1,58 @@
+module Fastlane
+  module Actions
+    class ZipAction < Action
+      def self.run(params)
+        UI.message "Compressing #{params[:path]}..."
+
+        params[:output_path] ||= "#{params[:path]}.zip"
+        sh "zip -r #{params[:output_path]} #{params[:path].shellescape}"
+
+        UI.success "Successfully generated zip file at path '#{File.expand_path(params[:output_path])}'"
+        return File.expand_path(params[:output_path])
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Compress a file or folder to a zip"
+      end
+
+      def self.details
+        "This zips everything in a weird way if you're not in the directory of the file to zip"
```

Comment:
You mean `Dir.chdir`? If so, do we have to `cd` to the source or the destination?